clusterctl/clusterdeployer/clusterclient.go

```diff
@@ -207,6 +207,50 @@ func (c *clusterClient) CreateMachineObjects(machines []*clusterv1.Machine) erro
 	return nil
 }
 
+// DeleteApiServerStackDeployments attempts to delete apiserver stack resources
+func (c *clusterClient) DeleteApiServerStackDeployments() error {
+	// delete cluster controller
+	cmd := exec.Command("kubectl", c.buildKubectlDeleteDeploymentArgs("deployment","clusterapi-controllers")...)
```

Comment:
These calls are making assumptions on the name and form of the controllers. This is not a safe assumption since d.providerComponents could have anything with any name. I would suggest the deletion not depend on specific names etc in providerComponents.

kubectl delete should be able to do that. https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#delete